api.js

```diff
@@ -86,16 +86,34 @@ exports.init = function (sbot, config) {
           details.feedformat,
           details.metadata
         )
-        const contentSection = [msgVal.content, msgVal.contentSignature]
-        const validationResult = validate.validateSingle(contentSection)
-        if (validationResult instanceof Error) return cb(validationResult)
-        sbot.db.add(msgVal, (err, msg) => {
+
+        const encrypted = typeof msgVal.content === 'string'
+
+        if (!encrypted) {
+          const contentSection = [msgVal.content, msgVal.contentSignature]
+          const validationResult = validate.validateSingle(contentSection)
+          if (validationResult instanceof Error) return cb(validationResult)
+        }
+
+        sbot.db.add(msgVal, (err, addedMsg) => {
           if (err) return cb(err)
-          const hydratedSubfeed = sbot.metafeeds.query.hydrateFromMsg(
-            msg,
-            metafeed.seed
-          )
-          cb(null, hydratedSubfeed)
+
+          function hydrate(msg) {
+            const hydratedSubfeed = sbot.metafeeds.query.hydrateFromMsg(
+              msg,
+              metafeed.seed
+            )
+            cb(null, hydratedSubfeed)
+          }
```

Comment:
Maybe we don't need this function because it's just a wrapper around `hydrateFromMsg` (after all both the wrapped and the wrapper are "hydrating from a message"). My main concern is memory usage and performance, because this function is created/defined for every `create()` call and it doesn't seem really necessary.

So on lines 111 and 116, just call hydrateFromMsg

Reply:
Thanks good idea.